### ISSUE TITLE
504Error(ImageMagick: v6.9.3-10, ngx_small_light: v0.7.1)

### ISSUE THREAD

**Issue by osamunmun** [cubicdaiya/ngx_small_light#41]
After updating ngx_small_light v0.6.14 to v0.7.1(I confirmed v0.7.0 OK), I got 504 error from nginx for some files. I use S3 as original image store. A part of nginx.conf is follows.

- I got 504 error 2 seconds later after access.
- Most files are OK, but some files are NG. I cannot provide the file. The one of the files is JPG and 942KB.
- I confirmed v0.7.0 is OK. I checked diff between v0.7.0 and v0.7.1, but I couldn't identify issue.

```
server {
    listen 80;
...
    location ~ ^/(.+)/(.+)/(.+)\.(jpg|jpeg|webp|png|gif)$ {
...
      set $param1   $1;
      set $param2   $2;
      set $filename $3;
      set $ext      $4;
      set $dw       '';
      set $dh       '';
      set $cw       '';
....
      proxy_pass http://127.0.0.1:8081/small_light(q=95,jpeghint=y,da=$da,dw=$dw,dh=$dh,cw=$cw,ch=$ch,of=$ext,cc=$cc)/img/$filename.$org;
    }
....
}

server {
    listen 8081 default_server;
    sendfile      on;
    tcp_nopush    on;
    small_light on;
    small_light_buffer 1m;
    send_timeout 120;
    access_log off;
    resolver 8.8.8.8 valid=5s;
    set_by_lua_file $bucket '/usr/local/nginx/lua/env.lua' 'BUCKET';
    set_by_lua_file $access_key '/usr/local/nginx/lua/env.lua' 'AWS_ACCESS_KEY';
    set_by_lua_file $secret_key '/usr/local/nginx/lua/env.lua' 'AWS_SECRET_KEY';

    location ~ ^/small_light[^/]*/(.+)$ {
        set $file $1;
        rewrite ^ /$file;
    }

    location ~ ^/img/(.+)$ {
        set $full_filename $1;
        rewrite_by_lua_file '/usr/local/nginx/lua/s3_auth.lua';
        proxy_hide_header x-amz-id-2;
        proxy_hide_header x-amz-request-id;
        proxy_hide_header Cache-Control;
        proxy_set_header Host $bucket.s3.amazonaws.com;
        proxy_pass http://$bucket.s3.amazonaws.com/$1;
        error_page 415 /50x.html;
    }
...
}
```

**Comment by cubicdaiya** [comment]
@osamunmun 

Is the phenomenon reproduced with simpler configuration like below?

```nginx
server {
    small_light on;
    location ~ small_light[^/]*/(.+)$ {
        set $file $1;
        rewrite ^ /$file;
    }

    location / {
        resolver 8.8.8.8;
        proxy_pass ${aws_s3_url};
    }
}
```

And what nginx version are you using?

**Comment by osamunmun** [comment]
@cubicdaiya 

I tried simpler configuration like below.
I tried this local environment, I didn't got 504 error. And I cloud measure response time.
v0.7.0 took about 1 sec, v0.7.1 took over 12 sec against same file.


```
server {
    small_light on;

    location ~ ^/small_light[^/]*/(.+)$ {
        set $file $1;
        rewrite ^ /$file;
    }

    location ~ ^/img/(.+)$ {
        resolver 8.8.8.8;
        proxy_pass http://xxxxxxxxxxxxxx.s3.amazonaws.com/$1;
    }
}
```

**Comment by cubicdaiya** [comment]
@osamunmun 

What nginx version are you using?  Though I tried with nginx-1.10.0 and ngx_small_light-0.7.1 and S3 in the configuration live above, it is not reproduced.

**Comment by cubicdaiya** [comment]
Hi, @osamunmun 

As I reproduced and fixed in 472348f, please check it.

**Comment by osamunmun** [comment]
@cubicdaiya Thanks a lot!
I have confirmed the bug fixed.

**Comment by cubicdaiya** [comment]
@osamunmun Thank you for the report. I released v0.7.2 included the bugfix.
